tests/unit/package_managers/test_npm.py

```diff
@@ -1378,3 +1587,166 @@ def args_based_return_checksum(integrity: str) -> ChecksumInfo:
+                "git+ssh://git@foo.org/foo-namespace/git-repo.git#YOLO1234": "deps/git-repo.git#YOLO1234.tgz",
+                "https://gitfoo.com/https-namespace/https-tgz/raw/tarball/https-tgz-3.0.0.tgz": "deps/https-tgz-3.0.0.tgz",
+            },
+            {
+                "lockfileVersion": 2,
+                "packages": {
+                    "": {
+                        "workspaces": ["foo", "bar"],
+                        "version": "1.0.0",
+                        "dependencies": {
+                            "@types/zzz": "^18.0.1",
+                            "hm-tarball": "",
+                            "git-repo": "",
+                        },
+                    },
+                    "node_modules/foo": {"version": "1.0.0", "resolved": "foo", "link": True},
+                    "node_modules/bar": {"version": "2.0.0", "resolved": "bar", "link": True},
+                    "node_modules/@yolo/baz": {
+                        "version": "0.16.3",
+                        "resolved": "file://${output_dir}/deps/baz-0.16.3.tgz",
+                        "integrity": "sha512-YOLO8888",
+                    },
+                    "node_modules/git-repo": {
+                        "version": "2.0.0",
+                        "resolved": "file://${output_dir}/deps/git-repo.git#YOLO1234.tgz",
+                        "integrity": "",
+                    },
+                    "node_modules/https-tgz": {
+                        "version": "3.0.0",
+                        "resolved": "file://${output_dir}/deps/https-tgz-3.0.0.tgz",
+                        "integrity": "sha512-YOLO-4321",
+                        "dependencies": {
+                            "@types/zzz": "^18.0.1",
+                            "hm-tarball": "",
+                            "git-repo": "",
+                        },
+                    },
+                    # Check that file dependency wil be ignored
+                    "node_modules/file-foo": {
+                        "version": "4.0.0",
+                        "resolved": "file://file-foo",
+                    },
+                },
+            },
+            id="update_package-lock_json",
+        ),
+    ],
+)
+def test_update_package_lock_with_local_paths(
+    rooted_tmp_path: RootedPath,
+    lockfile_data: dict[str, Any],
+    download_paths: Dict[NormalizedUrl, RootedPath],
+    expected_lockfile_data: dict[str, Any],
+) -> None:
+    for url, download_path in download_paths.items():
+        download_paths.update({url: rooted_tmp_path.join_within_root(download_path)})
+    package_lock = PackageLock(rooted_tmp_path, lockfile_data)
+    update_package_lock_with_local_paths(download_paths, package_lock)
+    assert package_lock.lockfile_data == expected_lockfile_data
+
+
+@pytest.mark.parametrize(
+    "file_data, workspaces, expected_file_data",
+    [
+        pytest.param(
+            {
+                "dependencies": {
```

Comment:
We're testing only the `dependencies` key in both package.json and package-lock.json, not the other 3. But we're planning to test some of the others in integration tests, so it's probably fine

Reply:
It would be a very easy fix to simply distribute the current dependencies into different keys (devDependencies, peerDependencies, optionalDependencies), though.

Reply:
Other dependency types added to unit test